### PR TITLE
initial Azure blob storage

### PR DESCRIPTION
**Pull request by discentem**
initial Azure blob storage

### CODE REVIEW

File: internal/stores/stores.go [hunk 1]
```diff
@@ -33,6 +34,15 @@ type Store interface {
...
 
+var (
+	// ensure AzureblobStore meets Store interface
+	_ = Store(&AzureBlobStore{})
+	// ensure S3Store meets Store interface
+	_ = Store(&S3Store{})
+	// ensure GCSStore meets Store interface
+	_ = Store(&GCSStore{})
+)
+
```
Comment (natewalck) [inline internal/stores/stores.go:45]: What is this for? I don't remember needing this for GCS.

File: internal/cli/init.go [hunk 2]
```diff
@@ -14,10 +14,18 @@ import (
...
+		Long:  fmt.Sprintf("Initialize a new %s repo", program.Name),
+		// Args:  cobra.ExactArgs(1),
+		Args: func(cmd *cobra.Command, args []string) error {
+			fn := cobra.ExactArgs(1)
+			err := fn(cmd, args)
+			if err != nil {
+				return errors.New("you must specify a path to a repo you want cavorite to track")
+			}
+			return nil
+		},
```
Comment (natewalck) [inline internal/cli/init.go:28]: I assume the default error here is too generic and not super helpful.

File: internal/cli/init.go [hunk 3]
```diff
@@ -14,10 +14,18 @@ import (
...
 func initCmd() *cobra.Command {
 	initCmd := &cobra.Command{
-		Use:     "init",
-		Short:   fmt.Sprintf("Initialize a new %s repo", program.Name),
-		Long:    fmt.Sprintf("Initialize a new %s repo", program.Name),
-		Args:    cobra.ExactArgs(1),
+		Use:   "init",
+		Short: fmt.Sprintf("Initialize a new %s repo", program.Name),
+		Long:  fmt.Sprintf("Initialize a new %s repo", program.Name),
+		// Args:  cobra.ExactArgs(1),
```
Comment (natewalck) [inline internal/cli/init.go:20]: Delete

**Review by natewalck — CHANGES_REQUESTED** [review]
One delete, the rest is context on the changes.

File: internal/cli/init.go (hunk 2 above)
Comment (discentem) [reply internal/cli/init.go:28]: yep exactly. default error was not helpful

File: internal/stores/stores.go (hunk 1 above)
Comment (discentem) [reply internal/stores/stores.go:45]: I don't think it's strictly necessary but it made it more obvious to me when I accidentally broke the interface implementations.

Comment (natewalck) [reply internal/stores/stores.go:45]: How does this help? I'm 100% not following 😭

**Review by natewalck — APPROVED** [review]
shipit7

**Pull request merged**